### ISSUE TITLE
Make running this easy for non devs - end users.

### ISSUE THREAD

**Issue by gedw99** [timelinize/timelinize#61]
Just so we don’t forget this from https://github.com/timelinize/timelinize/pull/58#issuecomment-2423848164

it’s so end users can run this easily off a cli and/or web view.

could embed a task file that does not have the dev time stuff like build , release etc .

The other thing will be a thing that makes it easy for users to configure their data sources .

I know this is an epic  btw.

Then it begs the question if the web gui  can reuse the same golang  that the task file uses .

like a htmx or datastar . Datastar is real time multi tab and has all the bells and whistles . https://github.com/delaneyj/datastar

**Comment by mholt** [comment]
I agree and recognize the current "get started" flow is only for tech-savvy users.

My eventual hope is to have:

- An installer for Windows
- A self-contained app package or binary for Linux (potentially that installs dependencies if root is not needed)
- A self-contained binary for macOS that installs dependencies on first run, maybe (I don't think root is required, I might be wrong) -- or even a DMG that you can unpack like "Traditional" mac apps.

**Comment by mohammed90** [comment]
Goreleaser can generate [MSI](https://goreleaser.com/customization/msi/) (Windows installer), [DMG](https://goreleaser.com/customization/dmg/) and [.app](https://goreleaser.com/customization/app_bundles/) (for Mac), and [Snaps](https://goreleaser.com/customization/snapcraft/) (for Ubuntu store), plus the various Linux packaging formats. Those advanced packages come with goreleaser Pro, which can be as low as ~$8 (https://goreleaser.com/pro/).

**Comment by mholt** [comment]
@gedw99 We now have CI producing automated builds at every push (https://github.com/timelinize/timelinize/actions/workflows/release.yml -- click the top job, then scroll down to find the asset download links), and we also have automated releases at every tag, which contain binaries: https://github.com/timelinize/timelinize/releases

I've also updated the documentation (and will continue to improve it!) reflecting the new instructions for users.

I'll close this now, since I think we're making progress on this, but feel free to continue discussion if needed.

(Proper platform-dependent installers as I mentioned above are still on my todo list, but that's another issue I think; and it probably necessitates code signing which is 💸 )

**Comment by mholt** [comment]
I'll look into it, but it's unlikely that we'll use a framework anytime soon (I'm still partial to vanilla JS).
